### PR TITLE
issue614 - added db-name option to select

### PR DESCRIPTION
**Pull request by Skaaal**
db-name command-line option filters the db dumper by connectionname
added tests based on 2 sqlite databases
composer test OK

### CODE REVIEW

File: src/Tasks/Backup/BackupJob.php [hunk 1]
```diff
@@ -51,6 +51,21 @@ public function dontBackupFilesystem(): self
         return $this;
     }
 
+    public function onlyDbName(array $dbNames): self
+    {
+        // Remove any DbDumper whose key does not equal any dbNames value
```
Comment (freekmurze) [inline src/Tasks/Backup/BackupJob.php]: Remove this comment.

File: src/Tasks/Backup/BackupJob.php [hunk 2]
```diff
@@ -51,6 +51,21 @@ public function dontBackupFilesystem(): self
...
+        $allowedNames = collect($dbNames);
+
+        $filteredDumpers = $this->dbDumpers->filter(
+            function ($value, $key) use ($allowedNames) {
```
Comment (freekmurze) [inline src/Tasks/Backup/BackupJob.php]: Rename `value` to `dbDumper` and `key` to `connectionName`

File: src/Tasks/Backup/BackupJobFactory.php [hunk 3]
```diff
@@ -24,8 +24,8 @@ protected static function createFileSelection(array $sourceFiles): FileSelection
 
     protected static function createDbDumpers(array $dbConnectionNames): Collection
     {
-        return collect($dbConnectionNames)->map(function (string $dbConnectionName) {
-            return DbDumperFactory::createFromConnection($dbConnectionName);
+        return collect($dbConnectionNames)->mapWithKeys(function (string $dbConnectionName) {
```
Comment (freekmurze) [inline src/Tasks/Backup/BackupJobFactory.php]: Nice!

File: tests/TestHelper.php [hunk 4]
```diff
@@ -43,6 +43,11 @@ public function getStubDirectory()
         return __DIR__.'/stubs';
     }
 
+    public function getStubDBDirectory()
```
Comment (freekmurze) [inline tests/TestHelper.php]: Rename this to `getSubDbDirectory`. Also add a `string` return type hint.

**Comment by Skaaal** [comment]
Done. That's my first contribution ever, so do not hesitate to suggest further improvements.

**Comment by freekmurze** [comment]
Thanks. I'll give it a second review soon.

And congratulations with your first contribution, you're awesome 👍

**Comment by freekmurze** [comment]
Thank you!

**Pull request merged**